binaries/scheduler/main.go

```
@@ -6,38 +6,97 @@ import (
+	bag := ice.NewMagicBag()
+	bag.PutMany(
+		func() (thrift.TServerTransport, error) { return thrift.NewTServerSocket(*addr) },
+		endpoints.MakeStatsReceiver,
+		func(s stats.StatsReceiver) *endpoints.ObservabilityServer {
+			return endpoints.NewObservabilityServer(fmt.Sprintf(":%d", *httpPort), s)
+		},
+		makeServers,
+		saga.MakeSagaCoordinator,
+		saga.MakeInMemorySagaLog,
+		distributor.NewPoolDistributorFromCluster,
+		scheduler.NewScheduler,
+		server.StartScheduler,
+		server.MakeServer,
+		thrift.NewTTransportFactory,
+		func() thrift.TProtocolFactory {
+			return thrift.NewTBinaryProtocolFactoryDefault()
+		},
+	)
+
+	o := jsonconfig.Options(map[string]jsonconfig.Interface{
+		"Cluster": {
+			"memory": &scootconfig.ClusterMemoryConfig{},
+			"local":  &scootconfig.ClusterLocalConfig{},
+			"": &scootconfig.ClusterMemoryConfig{
+				Type:  "memory",
+				Count: 10,
+			},
+		},
+		"Queue": {
+			"memory": &scootconfig.QueueMemoryConfig{},
+			"": &scootconfig.QueueMemoryConfig{
+				Type:     "memory",
+				Capacity: 1000,
+			},
+		},
+		"Workers": {
+			"local": &scootconfig.WorkersLocalConfig{},
+			"rpc":   &scootconfig.WorkersThriftConfig{},
+			"":      &scootconfig.WorkersLocalConfig{Type: "local"},
+		},
+	})
 
-	// Construct scootapi server handler based on config.
-	handler, err := parser.Create([]byte(*cfgText))
+	mod, err := o.Parse([]byte(*cfgText))
 	if err != nil {
 		log.Fatal("Error configuring Scoot API: ", err)
 	}
+	bag.InstallModule(mod)
 
-	// Start API Server
-	log.Println("Starting API Server")
-
-	err = server.Serve(handler, *addr,
-		thrift.NewTTransportFactory(),
-		thrift.NewTBinaryProtocolFactoryDefault())
+	var servers servers
+	err = bag.Extract(&servers)
 	if err != nil {
-		log.Fatal("Error serving Scoot API: ", err)
+		log.Fatal("Error injecting servers", err)
 	}
+
+	// TODO(dbentley): if one fails and the other doesn't, we should do
```

how does one go routine fail? If one panics the whole thing crashes right?

I think I'd prefer for safety to fail fast and hard.  So just crash the whole thing.  Aurora will restart it